Document Code Source links (#505)
pydantic/logfire: docs/guides/advanced/link-to-code-source.md

!!! warning "Experimental"
    This feature is experimental and may change in the future.

>     Please give us feedback on [Slack][slack].

Reviewer: Does it make sense to have an `experimental` namespace like we do in `pydantic` for things like this that might explicitly change?

Author: Answered in person.

> We support linking to the source code on GitHub, or any other VCS provider.

Reviewer: ```suggestion
We support linking to the source code on GitHub or any other VCS provider.
```

![Link to GitHub](../../images/guide/link-to-github.gif)

## Usage

> To enable source code linking with Logfire, you need to configure your project. Here's how:

Reviewer: ```suggestion
To enable source code linking with Logfire, you need to configure your project's code source. Here's an example:
```

Reviewer 2: ```suggestion
Here's an example:
```

This configuration is not at the project level anyway.

```python
import logfire

logfire.configure(
    code_source=logfire.CodeSource(

>         repository='https://github.com/pydantic/logfire',

Reviewer 2: This (combined with "We support linking to the source code on GitHub, or any other VCS provider") makes it look like it'll just work if they provide a non-github link, which it likely won't.

Author: What do you suggest?

Reviewer 2: It looks like our URL construction strategy works for gitlab (e.g. https://gitlab.com/rluna-gitlab/gitlab-ce/blob/v16.0.10/yarn.lock#L13) so i guess we support that too? maybe we just say we support github, gitlab, and any other VCS provider that uses the same URL format? we clearly don't support ANY provider.

Author: Ah, ok. I see.

Author: changed

>         revision='main',

Reviewer 2: This is usually not good practice and will lead to stale links, we should at least point this out.

I think ideally there would be a way to say 'use the service_version as the revision' and then we'd also encourage them to use service versions.

Author: better now?

>         root_path='.',

Reviewer 2: Should this just be the default?

Author: I can change this in a future PR, but I guess it makes sense to be the default yes.

Reviewer 2: seems like it should be optional in the data class but the frontend needs to handle defaulting to it so that the env var isn't required

Author: frontend does handle default already

    )
)
```

> You can check more about it on the [`logfire.CodeSource`][logfire.CodeSource] on our API reference.

Reviewer: ```suggestion
You can learn more in our [`logfire.CodeSource`][logfire.CodeSource] API reference.
```

> | Parameter | Description |

Reviewer 2: This table feels redundant with the example. What about adding comments to the example? Maybe those clickable bubbles?

Author: yes, bubbles.

| --- | --- |
| `repository` | The URL of the repository e.g. `https://github.com/pydantic/logfire` |
| `revision` | The specific branch, tag, or commit hash to link to e.g. `main` |
| `root_path` | The path to the root of the repository. If your code is in a subdirectory, you can specify it here. |

> ## Alternative Configuration

Reviewer: Seems like this pattern comes up a lot (re our settings vs otel env vars). Maybe we should start using tab panels to explain how to set things (one tab for logfire, one tab for otel env var)?

Author: Yes, that would be great.

> You can configure these settings using the [`OTEL_RESOURCE_ATTRIBUTES`][otel-resource-attributes] environment variable:

Reviewer: ```suggestion
Alternatively, you can configure these settings using the [`OTEL_RESOURCE_ATTRIBUTES`][otel-resource-attributes] environment variable:
```

> Alternatively, you can configure these settings using the [`OTEL_RESOURCE_ATTRIBUTES`][otel-resource-attributes] environment variable:

Reviewer 2: ```suggestion
For other OpenTelemetry SDKs, you can configure these settings using resource attributes, e.g. by setting the [`OTEL_RESOURCE_ATTRIBUTES`][otel-resource-attributes] environment variable:
```
